plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/expression/ExpressionConverter.java

```diff
@@ -0,0 +1,260 @@
+        return new NameReference(columnName);
+    }
+
+    private Expression convertTrinoConstant(Constant trinoConst)
+    {
+        Object value = trinoConst.getValue();
+        Type constantType = HudiTypeConverter.getType(trinoConst.getType());
+        Object literal;
+        if (value instanceof Slice slice) {
+            // Convert slice to string
+            literal = slice.toStringUtf8();
+        }
+        else {
+            literal = value;
+        }
+        return new Literal<>(literal, constantType);
+    }
+
+    private Expression convertTrinoCall(Call trinoCall)
+    {
+        FunctionName functionName = trinoCall.getFunctionName();
+        List<Expression> convertedArguments = new ArrayList<>();
+        for (ConnectorExpression arg : trinoCall.getArguments()) {
+            convertedArguments.add(convertExpressionRecursive(arg));
+        }
+
+        return mapTrinoFunctionToCustomExpression(functionName, convertedArguments, trinoCall.getType());
+    }
+
+    private Expression mapTrinoFunctionToCustomExpression(
+            FunctionName functionName, List<Expression> arguments, io.trino.spi.type.Type trinoType)
+    {
+        Type hudiReturnType = HudiTypeConverter.getType(trinoType);
+
+        Expression finalHudiExpression;
+
+        // Logical Operators
+        if (StandardFunctions.AND_FUNCTION_NAME.equals(functionName)) {
+            if (arguments.isEmpty()) {
+                // Handle case of empty arguments for AND if necessary
+                throw new IllegalArgumentException("AND function expects at least one argument.");
+            }
+            if (arguments.size() == 1) {
+                // AND(X) is just X
+                finalHudiExpression = arguments.getFirst();
+            }
+            // Iteratively build the AND chain
+            Expression result = arguments.get(0);
+            for (int i = 1; i < arguments.size(); i++) {
+                result = Predicates.and(result, arguments.get(i));
+            }
+            finalHudiExpression = result;
+        }
+        else if (StandardFunctions.OR_FUNCTION_NAME.equals(functionName)) {
+            if (arguments.isEmpty()) {
+                // Handle case of empty arguments for OR if necessary
+                throw new IllegalArgumentException("OR function expects at least one argument.");
+            }
+            if (arguments.size() == 1) {
+                // OR(X) is just X
+                return arguments.getFirst();
+            }
+            // Iteratively build the OR chain
+            Expression result = arguments.getFirst();
+            for (int i = 1; i < arguments.size(); i++) {
+                result = Predicates.or(result, arguments.get(i));
+            }
+            finalHudiExpression = result;
+        }
+        else if (StandardFunctions.NOT_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            finalHudiExpression = Predicates.not(arguments.getFirst());
+        }
```

Comment:
nit: pull `arguments.size() == 1` out of the condition check and put it into `Preconditions#checkArgument`; similar for other `else if` branches.